File: clear_page_app/lib/models/summary.dart
class Summary {
  final String coreExplanation;
  final String terminologyBreakdown;
  final String practicalUnderstanding;
  final String clarifications;
  final String mentalModel;

  Summary({
    required this.coreExplanation,
    required this.terminologyBreakdown,
    required this.practicalUnderstanding,
    required this.clarifications,
    required this.mentalModel,
  });

  factory Summary.fromJson(Map<String, dynamic> json) {
    return Summary(
      coreExplanation: json['coreExplanation'] ?? 'N/A',
      terminologyBreakdown: json['terminologyBreakdown'] ?? 'N/A',
      practicalUnderstanding: json['practicalUnderstanding'] ?? 'N/A',
      clarifications: json['clarifications'] ?? 'N/A',
      mentalModel: json['mentalModel'] ?? 'N/A',
    );
  }
}

File: backend/src/main/java/com/clearpage/backend/model/Summary.java
package com.clearpage.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
public class Summary {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // A hash of the content to quickly find if it's already summarized
    @Column(columnDefinition = "TEXT")
    private String contentHash;

    @Column(columnDefinition = "TEXT")
    private String originalText;

    @Column(columnDefinition = "TEXT")
    private String coreExplanation;

    @Column(columnDefinition = "TEXT")
    private String terminologyBreakdown;

    @Column(columnDefinition = "TEXT")
    private String practicalUnderstanding;

    @Column(columnDefinition = "TEXT")
    private String clarifications;

    @Column(columnDefinition = "TEXT")
    private String mentalModel;

    private LocalDateTime createdAt = LocalDateTime.now();
}
